bundles/org.openhab.binding.iammeter/src/main/java/org/openhab/binding/iammeter/internal/IammeterHandler.java

```diff
@@ -0,0 +1,149 @@
+import java.util.concurrent.TimeUnit;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.cache.ExpiringCache;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link IammeterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Yang Bo - Initial contribution
+ */
+
+@NonNullByDefault
+public class IammeterHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(IammeterHandler.class);
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private IammeterConfiguration config;
+    private static final int TIMEOUT_MS = 5000;
+    private final ExpiringCache<Boolean> refreshCache = new ExpiringCache<>(Duration.ofSeconds(5), this::refresh);
+
+    public IammeterHandler(Thing thing) {
+        super(thing);
+        config = getConfiguration();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            refreshCache.getValue();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        config = getConfiguration();
+        if (refreshJob == null) {
+            refreshJob = scheduler.scheduleWithFixedDelay(this::refresh, 0, config.refreshInterval, TimeUnit.SECONDS);
+            this.refreshJob = refreshJob;
+            updateStatus(ThingStatus.UNKNOWN);
+        }
+    }
+
+    private boolean refresh() {
+        refreshCache.invalidateValue();
+        IammeterConfiguration config = this.config;
+        try {
+            String httpMethod = "GET";
+            String url = "http://" + config.username + ":" + config.password + "@" + config.host + ":" + config.port
+                    + "/monitorjson";
+            String content = "";
+            InputStream stream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+
+            String response = HttpUtil.executeUrl(httpMethod, url, stream, null, TIMEOUT_MS);
```

Comment:
The `stream` is an empty String. Do you need to send something or does it also work to use:
```suggestion
            String response = HttpUtil.executeUrl(httpMethod, url, TIMEOUT_MS);
```
Same applies to the other handler

Reply:
both changed.